tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -173,7 +170,24 @@ impl<F, T> Format<F, T> {
     #[cfg_attr(docsrs, doc(cfg(feature = "json")))]
     pub fn json(self) -> Format<Json, T> {
         Format {
-            format: PhantomData,
+            format: Json::default(),
+            timer: self.timer,
+            ansi: self.ansi,
+            display_target: self.display_target,
+            display_level: self.display_level,
+        }
+    }
+
+    /// Use the full JSON format with event metadata flattened.
+    ///
+    /// See [`Json`].
+    #[cfg(feature = "json")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "json")))]
+    pub fn json_flatten(self) -> Format<Json, T> {
```

Comment:
I think that this should be in an impl block where the formatter type parameter is `Json`, like:

```rust
impl<T> Format<Json, T> {
   ...
}
```
and mutate the existing `Json` instance. That way, if we add more functions for controlling the `Json` formatter's behavior, we'll be able to expose them too.